### DISCUSSION TITLE
Time zone of shared Exchange calendar is displayed incorrectly

### DISCUSSION THREAD

**Discussion by BSG2000** [bitfireAT/icsx5#74] [Q&A] [answered]
I have an Exchange calendar in which I share appointments with my girlfriend. Since she only needs read access, we have solved this by sharing the calendar via ICS. Unfortunately, since the last time change, the serial appointments are displayed one hour later (too early). For example, instead of 17:00 in reality, they start at 16:00. Is there any way to fix this? Does anyone know about this problem?

You could provide an offset in the app configuration. That would be a workaround.

**Comment by ArnyminerZ** [comment]
If I'm not wrong, the Timezone is provided by the server. Please, make sure you have specified it when creating the events, and the app should adapt automatically. For example, in Google Calendar the Timezone is set using this field:

![Screenshot_20221212-152707~2](https://user-images.githubusercontent.com/12086466/207070670-4a31c8a2-3db2-4eff-9b3a-e8134361324c.png)

**Comment by sochotnicky** [comment]
I am actually having this **exact** problem with Exchange/Outlook. Recurring events are off by an hour. They are scheduled in a different timezone (US EST) and I am in CET (not sure whether that makes a difference here).

I checked and the shared ICS looks correct with event start like this:

    DTSTART;TZID=US Eastern Standard Time:20220418T110000

Which is 5pm CET but actually shows up at 6pm CET in my calendar. 

That same event has:

RRULE:FREQ=WEEKLY;UNTIL=20240415T150000Z;INTERVAL=1;BYDAY=MO;WKST=SU
EXDATE;TZID=US Eastern Standard Time:20221226T110000

So I wonder if icsx is taking the Timezone info from RRULE for a specific event. That would explain the one hour off (DST vs not)

**Comment by lsd0203** [comment]
I have the same problem with serial appointments (ics-subscription from Exchange server at work):
If the series begins in standard time zone (UTC+1), all serial appointments within date range of DST (UTC +2, ca. April to October) will be displayed at the wrong time (+1 hour, e.g. 14:00 h instead of 13:00 h). 
If I change the first appointment of the series to any day within DST (UTC +2), all appointments from ca. April to October will be displayed correctly, but starting with standard time zone (ca. November to March), it will be 1 hour too early (-1 hour, eg. 11:00 h instead of 12:00 h).

It can easily be reproduced by changing the first appointment of a serial appointment in Outlook/Exchange and synchronizing with icsx5 afterwards. Just try starting the series either March 20th (standard time) or March 27th 2023 (DST). 

Could you fix this issue concerning all serial appointments ranging over clock changing dates or do you need an issue entry for this?

**Comment by rfc2822** [comment]
Can you please provide

- an ICS file that causes the problem
- further information like Android version, your used timezone and detailled steps to reproduce the problem

so that we can have a look?

**Comment by lsd0203** [comment, marked as answer]
Hi Ricki,

 

actually, the issue no longer occurs. I think Microsoft changed something on the client (Office 365) or server side.

 


Best regards

Lukas

 
 

Gesendet: Samstag, 22. Juli 2023 um 13:08 Uhr
Von: "Ricki Hirner" ***@***.***>
An: "bitfireAT/icsx5" ***@***.***>
Cc: "lsd0203" ***@***.***>, "Comment" ***@***.***>
Betreff: Re: [bitfireAT/icsx5] Time zone of shared Exchange calendar is displayed incorrectly (Discussion #74)


 

Can you please provide


	an ICS file that causes the problem
	further information like Android version, your used timezone and detailled steps to reproduce the problem


so that we can have a look?

**Comment by mcvoss** [comment]
Hello,

As of today my calendar is still having the same issue. I am in Central Time zone, If an invite is created in Eastern Time Zone then the invite on my calendar reflects eastern time not central. Do I need to uninstall and reinstall ICSx5 to get Microsoft's fix to work?

Thanks,

**Reply by rfc2822** [reply]
Can you please provide steps to reproduce the problem and provide the .ics file required for it?

**Reply by mcvoss** [reply]
Here is the ics file. FYI, it's only on calendar invites that are created outside of my timezone. 
One thing I noticed after going through my calendar for the next 60 days, some reflect the timezone the appointment was created in and some don't. I also installed Onecaledar which uses the same ics file and it works fine. I prefer ICS though.
Thanks for the quick response.

Michael

**Reply by rfc2822** [reply]
Thanks for the ics file. Which exact steps do I have to do to reproduce the problem? I have to reproduce the problem so that I can have a look at it.

**Comment by mcvoss** [comment]
Hi Ricki,
All I did was enter the ics webcal address in ICSx5. Then I opened up a couple different calendar apps on my phone and added it as shared calendar. I hope that makes sense. Agin it's the same issue as some metiond above in the thread. I am running the lastest ICSx app. I wonder if uninstall and reinstall would work?
    On Thursday, July 27, 2023 at 12:44:11 PM CDT, Ricki Hirner ***@***.***> wrote:  
 
 


Thanks for the ics file. Which exact steps do I have to do to reproduce the problem? I have to reproduce the problem so that I can have a look at it.

**Reply by rfc2822** [reply]
I have now synchronized your ics file with ICSx5. What do I have to do now to see the problem? What is the actual problem? Please explain what to do now, what I then will see and what you expect that should be shown instead.

**Comment by mcvoss** [comment]
Ricki,

If you can see my calendar you will notice that there is an entry for today called "Schlage Lock Construction Check In". That entry show up as 11:30 am (GMT-4) That is the time zone that the person created the entry is in. I am in GMT-5 but the invite shows the time in GMT-4 for me. This only happens  sometimes when I get an invite from a different time zone.
When I go to my calendar in outlook the times are correct.
    On Thursday, July 27, 2023 at 01:19:16 PM CDT, Ricki Hirner ***@***.***> wrote:  
 
 


I have now synchronized your ics file with ICSx5. What do I have to do now to see the problem? What is the actual problem? Please explain what to do now, what I then will see and what you expect that should be shown instead.

**Reply by rfc2822** [reply]
> If you can see my calendar you will notice that there is an entry for today called "Schlage Lock Construction Check In".

It has this start time in the iCalendar:

> DTSTART;TZID=US Eastern Standard Time:20230727T113000 

As far as I see, "US Eastern Standard Time" is UTC-5.

When I set my calendar app to UTC-5 home time zone, the entry is shown as 11:30, as I'd expect.
When I set my calendar app to UTC-4 home time zone, the entry is shown as 12:30, as I'd expect.
